opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/AsyncDefaultAutoConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package io.opentracing.contrib.spring.cloud.async;
+
+import io.opentracing.Tracer;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
+import org.springframework.scheduling.annotation.AsyncConfigurer;
+import org.springframework.scheduling.annotation.AsyncConfigurerSupport;
+import org.springframework.scheduling.annotation.EnableAsync;
+
+import java.util.concurrent.Executor;
+
+/**
+ * @author kamesh
+ */
+@Configuration
+@EnableAsync
```

Comment:
should we enable async? I think user should decide to enable/disable spring features.

Reply:
The same question, although sleuth enables it's done here.

Reply:
i had the same doubt when doing it, Sleuth does it so added it but then I had another thought is that unless `@Async` added this will not be triggered, so when I do `@Async` then I need to `@EnabledAsync`